docs/asciidoc/static/configuration.asciidoc

```diff
@@ -64,74 +52,75 @@ input {
 
-The documentation for a plugin may enforce a configuration field having a
-certain type.  Examples include boolean, string, array, number, hash,
-etc.
-[[boolean]]
 [float]
-==== Boolean
-
-A boolean must be either `true` or `false`. Note the lack of quotes around `true` and `false`.
+=== Value Types
 
-Examples:
+A plugin can require that the value for a setting be a
+certain type, such as boolean or hash. The following value
+types are supported.
 
-[source,js]
-----------------------------------
-  ssl_enable => true
-----------------------------------
-[[string]]
+[[array]]
 [float]
-==== String
+==== Array
 
-A string must be a single value.
+An array can be a single string value or multiple values. If you specify the same
+setting multiple times, it appends to the array.
 
 Example:
 
 [source,js]
 ----------------------------------
-  name => "Hello world"
+  path => [ "/var/log/messages", "/var/log/*.log" ]
+  path => "/data/mysql/mysql.log"
 ----------------------------------
 
-You should use quotes around string values.
-[[number]]
+This example configures `path` to be an array that contains an element for each of the three strings.
+
+
+[[boolean]]
 [float]
-==== Number
+==== Boolean
 
-Numbers must be valid numerics (floating point or integer are OK).
+A boolean must be either `true` or `false`. Note that the `true` and `false` keywords
+are not enclosed in quotes.
 
 Example:
 
 [source,js]
 ----------------------------------
-  port => 33
+  ssl_enable => true
 ----------------------------------
-[[array]]
+
+[[codec]]
 [float]
-==== Array
+==== Codec
 
-An array can be a single string value or multiple. If you specify the same
-field multiple times, it appends to the array.
+A codec is the name of Logstash codec used to represent the data. Codecs can be
+used in both inputs and outputs.
 
-Examples:
+Input codecs provide a convenient way to decode your data before it enters the input.
+Output codecs provide a convenient way to encode your data before it leaves the output.
+Using an input or output codec eliminates the need for a separate filter in your Logstash pipeline.
+
+Example:
 
 [source,js]
 ----------------------------------
-  path => [ "/var/log/messages", "/var/log/*.log" ]
-  path => "/data/mysql/mysql.log"
+  codec => "json"
 ----------------------------------
 
-The above makes 'path' a 3-element array including all 3 strings.
 [[hash]]
 [float]
 ==== Hash
 
-A hash is basically the same syntax as Ruby hashes.
-The key and value are simply pairs, such as:
+A hash is a collection of key value pairs. Hashes are specified using the Ruby hash
```

Comment:
+1 on this improvement. The old version and this one are slightly inaccurate, I dont' know if we say elsewhere, but hashes are "ruby hash" syntax without commas.